torch/fx/traceback.py

```diff
@@ -15,6 +15,43 @@
 @compatibility(is_backward_compatible=False)
 @contextmanager
 def preserve_node_meta():
+    """
+    Preserves state of ``meta`` field of a graph node
+
+    This function preserves a node's meta fields (stack_trace, nn_module_stack, val, tensor_meta, ...)
+    during fx.Transformer and aot_autograd transformations.
+
+    Example:
+
+    >>> import torch
+
+    >>> class SimpleAdd(torch.nn.Module):
+    >>>     def __init__(self):
+    >>>         super().__init__()
+    >>>     def forward(self, x, y):
+    >>>         return x + y
+
+    >>> model = SimpleAdd()
+    >>> with torch.fx.traceback.preserve_node_meta():
+    >>>     fxg = torch.fx.symbolic_trace(model)
+
+    >>>     # node meta prior to transform
+    >>>     for node in fxg.graph.nodes:
+    >>>         print(node, node.meta)
+
+    >>>     def simple_transform(g: torch.fx.Graph) -> torch.fx.Graph:
```

Comment:
hmm, this is not really the transformation I was referring to. It should be with `fx.Transformer`.